runtime/src/main/java/org/corfudb/runtime/view/LayoutBuilder.java

```diff
@@ -328,37 +330,50 @@ public LayoutBuilder mergePreviousSegment(int segmentIndex) {
 
     /**
      * Moves a responsive server to the top of the sequencer server list.
+     * Tries to avoid electing a log tail node if possible to have better load distribution.
      * If all have failed, throws exception.
      *
-     * @param endpoints a non null set of Strings representing failed endpoints
-     *
+     * @param excludedNodes a non null set of Strings representing excluded endpoints
      * @return this builder
-     *
-     * @throws LayoutModificationException is thrown if none of the sequencers in layout
-     *         can be moved to the top of sequencer server list due to being unresponsive
-     *         or a failed endpoint.
+     * @throws LayoutModificationException if none of the sequencers in the layout can be
+     *                                     moved to the top of sequencer server list due
+     *                                     to being unresponsive or a failed endpoint.
      */
-    public LayoutBuilder assignResponsiveSequencerAsPrimary(@NonNull Set<String> endpoints) {
-
-        List<String> modifiedSequencerServers = new ArrayList<>(layout.getSequencers());
-        for (int i = 0; i < modifiedSequencerServers.size(); i++) {
-            String sequencerServer = modifiedSequencerServers.get(i);
-
-            // Add a sequencer server given sequencerServer is not already included in
-            // endpoints AND it is not included in the unresponsive nodes of the layout.
-            if (!endpoints.contains(sequencerServer) &&
-                !layout.getUnresponsiveServers().contains(sequencerServer)) {
-                modifiedSequencerServers.remove(sequencerServer);
-                modifiedSequencerServers.add(0, sequencerServer);
-                layout.setSequencers(modifiedSequencerServers);
-                return this;
+    public LayoutBuilder assignResponsiveSequencerAsPrimary(@NonNull Set<String> excludedNodes) {
+        List<String> modifiedSequencers = new LinkedList<>(layout.getSequencers());
+        String primarySequencer = null;
```

Comment:
can you please use Optional instead of null

Reply:
Done.